### PR TITLE
Migrate StyledOcticon to TypeScript

### PR DESCRIPTION
**Pull request by colebemis**
This PR migrates the `StyledOcticon` component to TypeScript as part of the [TypeScript refactor](https://github.com/primer/components/issues/970).

### CODE REVIEW

File: src/__tests__/StyledOcticon.tsx [hunk 1]
```diff
@@ -15,10 +15,6 @@ describe('StyledOcticon', () => {
     default: StyledOcticon
   })
 
-  it('implements system props', () => {
-    expect(StyledOcticon).toImplementSystemProps(COMMON)
-  })
```
Comment (colebemis) [inline src/__tests__/StyledOcticon.tsx:20]: This test is run in the `behavesAsComponent()` test helper.

File: src/StyledOcticon.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+StyledOcticon.defaultProps = {
+  theme,
+  size: 16
+}
+
+StyledOcticon.propTypes = {
+  ...COMMON.propTypes,
+  ...sx.propTypes,
+  icon: PropTypes.any.isRequired,
+  size: PropTypes.any,
```
Comment (colebemis) [inline src/StyledOcticon.tsx:30]: These propTypes were causing errors. Since we'll be removing propTypes soon, I think `any` is fine for now.

Comment (smockle) [reply src/StyledOcticon.tsx:30]: I’d personally go ahead and remove PropTypes from this file, but this is fine too!

**Review by smockle — APPROVED** [review]
🎊

**Pull request merged**